OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java

```diff
@@ -1952,6 +1952,26 @@ public static boolean isLastWordCityGroup(SearchPhrase p ) {
 				p.isLastWord(ObjectType.VILLAGE);
 	}
 
+	public static SearchResult createAmenitySearchResult(Amenity amenity) {
```

Comment:
It looks complicated, do we need all these? or is it enough to create similar to [this](https://github.com/osmandapp/Osmand/blob/2c5b8b4394ab06ece5ed00ac817c729b8740a4fa/OsmAnd/src/net/osmand/plus/search/QuickSearchHelper.java#L418-L426)?